docs/dev.rst

```diff
@@ -1,20 +1,513 @@
+
+You can also include/exclude components from the list using the include/exclude options:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev --include "test.*,another_regex" --exclude "something_to_exclude"
+
+The ``include`` and ``exclude`` patterns are matched against both the metadata type and name of the component.
+
+You can also include all changed components of specific types:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev --types "CustomObject,CustomField"
+
+
+
+Retrieve Changes
 ----------------
+The ``retrieve_changes`` task supports both ``sfdx`` and ``mdapi`` formatted source code. 
+It also utilizes the `SourceMember <https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_sourcemember.htm>`_ sObject to detect what has changed in an org,
+but allows you to be more selective regarding which components to retrieve when compared to the ``dx_pull`` task. 
+Manual tracking of component versions also allows for the possibility of retrieving some changes into one directory,
+and then running the task again to retrieve other changes into a different directory.
+ 
+.. note::
+
+    CumulusCI has multiple tasks for retrieving Metadata from an org environment.
+    For a comprehensive list, see the list of :ref:`retrieve changes` tasks in the cheat sheet.
 
-Track and retrieve changes
---------------------------
+When you are ready to capture changes in an org, run the ``retrieve_changes`` task:
 
-Run Apex tests
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev
+
+The task accepts ``include``, ``exclude``, and ``types`` options for filtering the
+list of changed components, for scenarios where you don't want to retrieve all changed components.
+
+After the metadata has been retrieved, the org snapshot will be updated so 
+that the retrieved components will no longer be included in ``list_changes``.
+You can avoid this by setting the ``snapshot`` option to False.
+
+By default, changes are retrieved into the ``src`` directory when using Metadata source format,
+or the default ``sfdx`` package directory (``force-app``) when using ``sfdx`` source format.
+You can retrieve into a different location using the ``path`` option:
+
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev --path unpackaged/config/qa
+
+
+
+Push Changes
+------------
+Developers rarely edit code directly in an org environment, but instead use an editor or IDE like VSCode or IntelliJ.
+After code (or other Metadata) in an editor you will need to push these changes from your project's local repository to the target org.
+
+If your project uses the ``sfdx`` source format then you can use the ``dx_push`` task:
+
+.. code-block:: console
+
+    $ cci task run dx_push
+
+If you project uses the Metadata source format you can use the ``deploy`` task:
```

Comment:
Let's not recommend using the deploy task on its own. There are things it doesn't do (unscheduling Apex classes so they can be updated, updating package.xml, deleting components that were removed from the source). Instead, we should recommend running the `deploy_unmanaged` flow. This supports both source formats (but dx_push will be more efficient in the case where only one file was changed, because deploy_unmanaged re-deploys the entire package)